polar-core/src/kb.rs

```diff
@@ -943,6 +967,26 @@ mod tests {
             )
             .unwrap()
             .is_true());
+        // rule: f(x: 6), rule_type: f(x: String) => FAIL
+        assert!(!kb
+            .rule_params_match(
+                &rule!("f", ["x"; value!(6)]),
+                &rule!("f", ["x"; instance!(sym!("String"))])
+            )
+            .unwrap()
+            .is_true());
```

Comment:
what's the deal with the new tests in here that are checking primitive values against built-in types w/o any fields? This one + `Float`, `Integer`, `Boolean`, `List`, and `Dictionary`

Reply:
I realized that we didn't test that the check would fail if the rule type specializer is a primitive and the rule value is the wrong type. We were only checking if the value was a primitive and the specializer was Foo, which would be handled by a different piece of code since the function matches on the type of the rule type rather than the rule.